stream/src/utils/mod.rs

```diff
@@ -0,0 +1,3 @@
+//! Utility functions for the stream module.
+
+pub mod codec;
```

Comment:
nit: Since there is only one type of `util`, I'd be open to removing this indirection (just putting this logic in a `utils.rs` file)?

I think what you have now is preferred but just flagging.

Reply:
will leave as-is for now